test/sanitize_json.d

```diff
@@ -0,0 +1,146 @@
+import std.exception : assumeUnique;
+import std.string : startsWith, replace;
+import std.format : formattedWrite, format;
+import std.json;
+import std.file : readText;
+import std.stdio;
+
+void usage()
+{
+    writeln("Usage: santize_json <input-json> [<output-json>]");
+}
+int main(string[] args)
+{
+    args = args[1 .. $];
+    if (args.length == 0)
+    {
+        usage();
+        return 1;
+    }
+    string inFilename = args[0];
+    File outFile;
+    if(args.length == 1)
+    {
+        outFile = stdout;
+    }
+    else if(args.length == 2)
+    {
+        outFile = File(args[1], "w");
+    }
+    else
+    {
+        writeln("Error: too many command line arguments");
+        return 1;
+    }
+
+    auto json = parseJSON(readText(inFilename));
+    sanitize(json.array);
+
+    outFile.write(json.toJSON(true));
+    return 0;
+}
+
+void sanitize(JSONValue[] rootArray)
+{
+    foreach (ref obj; rootArray)
+    {
+        auto kind = obj.object["kind"].str;
+        if (kind == "compilerInfo")
+            sanitizeCompilerInfo(obj.object);
+        else if (kind == "buildInfo")
+            sanitizeBuildInfo(obj.object);
+        else if(kind == "module")
+            sanitizeSyntaxNode(obj);
+        else if(kind == "semantics")
+            sanitizeSemantics(obj.object);
+    }
+}
+
+void removeString(JSONValue* value)
+{
+    assert(value.type == JSON_TYPE.STRING);
+    *value = JSONValue("VALUE_REMOVED_FOR_TEST");
+}
+void removeNumber(JSONValue* value)
+{
+    assert(value.type == JSON_TYPE.INTEGER || value.type == JSON_TYPE.UINTEGER);
+    *value = JSONValue(0);
+}
+void removeStringIfExists(JSONValue* value)
+{
+    if (value !is null)
+        removeString(value);
+}
+
+void sanitizeCompilerInfo(ref JSONValue[string] buildInfo)
+{
+    removeString(&buildInfo["binary"]);
+    removeString(&buildInfo["version"]);
+}
+void sanitizeBuildInfo(ref JSONValue[string] buildInfo)
+{
+    removeString(&buildInfo["cwd"]);
+    removeStringIfExists("config" in buildInfo);
+    removeStringIfExists("lib" in buildInfo);
+    {
+        auto importPaths = buildInfo["importPaths"].array;
+        foreach(ref path; importPaths)
+        {
+            path = JSONValue(normalizeFile(path.str));
+        }
+    }
+}
+void sanitizeSyntaxNode(ref JSONValue value)
+{
+    if (value.type == JSON_TYPE.ARRAY)
+    {
+        foreach (ref element; value.array)
+        {
+            sanitizeSyntaxNode(element);
+        }
+    }
+    else if(value.type == JSON_TYPE.OBJECT)
+    {
+        foreach (name; value.object.byKey)
+        {
+            if (name == "file")
+                removeString(&value.object[name]);
+            else if (name == "deco")
+                removeString(&value.object[name]);
+            else if (name == "offset")
+                removeNumber(&value.object[name]);
+            else
+                sanitizeSyntaxNode(value.object[name]);
```

Comment:
Not a fan of this (I know that's how the status quo is).
Ideally this would be configurable via a flag, s.t. special platform-specific, short tests which test that the `deco` is correctly set in the json output can be added, but that's definitely not something which shouldn't block this PR.

Reply:
Ok, I added a `--keep-deco` flag that can be used by tests in the future.